pkg/controller/add.go

```diff
@@ -54,10 +61,40 @@ func AddToManager(ctx context.Context, mgr manager.Manager) error {
 	return AddToManagerWithOptions(ctx, mgr, &DefaultAddOptions)
 }
 
+// watchInfrastructure watches for Infrastructure changes and triggers the Extension reconciliation.
+func watchInfrastructure(mgr manager.Manager) extensionscontroller.WatchBuilder {
+	infraStatusChangedPredicate := predicate.TypedFuncs[*extensionsv1alpha1.Infrastructure]{
```

Comment:
This predicate reacts on `Create` and `Delete` events, which is not what we want. Please implement `{Create,Delete}Func` and explicitly return `false`.